hyper-capi/src/lib.rs

```diff
@@ -21,7 +21,14 @@ pub enum IterStep {
     Break,
 }
 
-
 struct AssertSendSafe<T>(T);
 
 unsafe impl<T> Send for AssertSendSafe<T> {}
+
+static VERSION_CSTR: &str = concat!(env!("CARGO_PKG_VERSION"), "\0");
+
+ffi_fn! {
+    fn hyper_version() -> *const u8 {
+        VERSION_CSTR.as_ptr()
+    }
```

Comment:
In my build this returns `0.0.0` - I presume that's not expected?

Reply:
It's expected at this very moment, since it's the value inside `hyper-capi/Cargo.toml`. This is one point I've been thinking about with regards to making this a separate crate, it's technically a different version than the `hyper` crate. I'll be writing up the pros and cons about that in a separate issue.

But I've also been wondering, since this is to help users be able to report exactly what version of stuff they used, I suppose it may be helpful to include a couple of the versions of the bigger dependencies in hyper as well (like `h2`, hyper's HTTP/2 implementation that is a sub crate).

Reply:
It's pretty ugly, but if i'm reading this correctly, semver allows you to specify the associated `hyper` version as metadata of the `hyper-api` version: `0.0.1+hyper.0.3.18`.

Reply:
In my case I'm interested in the conveying of version information to users so that users subsequently can tell us that info back in a future when we're debugging weird things and we're curious what versions the user is actually using. "semver" has little value in that context. But as @seanmonstar points out, the version numbers of the most important sub-dependencies can certainly have value.

Reply:
I was under the impression that this would always be `hyper-api`'s crate version, which has to adhere to semver. If it can be anything, all the better.

Reply:
I've written up those thoughts into an [this issue](https://github.com/hyperium/hyper/issues/2299) (the version string is relevant, but a smaller point to what I wrote up).